### PR TITLE
Added base path support + minor refactoring

### PR DESCRIPTION
**Pull request by satya164**
Currently the plugin adds relative paths to the manifest file. Say I do the following,

```javascript
return gulp.src("public/s/**/*"])
          .pipe(appcache(options))
```

All, the file paths in the manifest file won't have "public/s/".

Say "public" is the folder which I serve, and not "public/s". Then the generated manifest file won't work for my configuration. If I can specify `options.basePath` as `public`, then this problem can easily be avoided, and it'll fit all use cases including mine.

The naming is similar to the `grunt-contrib-manifest` plugin.

Note that this pull request changes the behavior of the plugin.

### CODE REVIEW

**Comment by vdswouter** [comment]
+1

**Comment by LaurensRietveld** [comment]
+1. would like base URL support

**Comment by bgirschig** [comment]
+1 That would be really nice

**Comment by hillmanov** [comment]
Sorry it has taken so long to get to this - 

If you can take care of some of the conflicts, as well as take the readme out of the package.json, I can merge it in. 

Thanks!

**Comment by satya164** [comment]
@hillmanov done

File: package.json [hunk 1]
```diff
@@ -16,19 +16,29 @@
...
     "gulp-util": "~2.2.6",
     "slash": "^0.1.1",
     "minimatch": "~2.0.1"
-  }
+  },
+  "bugs": {
+    "url": "https://github.com/hillmanov/gulp-manifest/issues"
+  },
+  "homepage": "https://github.com/hillmanov/gulp-manifest",
+  "_id": "gulp-manifest@0.0.5",
```
Comment (hillmanov) [inline package.json]: Not sure where the rest of the meta data is coming from, but some of it isn't even correct. The npm version is now at 0.0.7, and will very soon be bumped. The shasum will be changed all the time, and the resolved thing will be too difficult to maintain. Can we drop this info as well?

**Comment by satya164** [comment]
@hillmanov I've no idea as well. I will remove it.

**Comment by satya164** [comment]
Done.

**Comment by satya164** [comment]
@hillmanov Any updates on this?

**Pull request merged**